packages/tocco-ui/src/EditableValue/typeEditors/DateAbstract.js

```diff
@@ -72,17 +73,19 @@ class DateAbstract extends React.Component {
     return null
   }
 
-  componentWillReceiveProps(props) {
-    this.handleButtonVisibility()
-    const locale = this.getLocale(props.intl.locale)
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    if (!_isEqual(prevProps.value, this.props.value)) {
+      this.handleButtonVisibility()
```

Comment:
does this have to be so complicated? Storing this simple value in store.

`
   <StyledDateAbstractControl
          data-clear	    
          hideButton={ !this.hasValue() && !this.props.readOnly}	          
        >	       
`

won't work?